clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -170,8 +229,11 @@ public void updateState(ConsumerGroupHeartbeatResponseData response) {
         ConsumerGroupHeartbeatResponseData.Assignment assignment = response.assignment();
         if (assignment != null) {
             setTargetAssignment(assignment);
+            transitionTo(MemberState.RECONCILING);
+            reconcile(targetAssignment.get());
```

Comment:
The IDE is showing this line with a warning because it's invoking `get()` without a `isPresent()` check. I know that it's being set in `setTargetAssignment` right above, but can we refactor this code to make it more obvious to the compiler (and any humans reading it)?

Here's a quick take:

 ```suggestion
            // Take new target assignment received from the server and set it as targetAssignment
            // to be processed. Following the consumer group protocol, the server won't send a
            // new target member while a previous one hasn't been acknowledged by the member, so
            // this will fail if a target assignment already exists.
            if (targetAssignment.isPresent()) {
                transitionToFailed();
                throw new IllegalStateException("Cannot set new target assignment because a " +
                    "previous one pending to be reconciled already exists.");
            }

            log.info("Member {} accepted new target assignment {} to reconcile", memberId, assignment);
            transitionTo(MemberState.RECONCILING);
            targetAssignment = Optional.of(assignment);
            reconcile(assignment);
```

That way all the logic is together and we can remove `setTargetAssignment()`, too. Just a thought.